### PR TITLE
Workflow for nightly kubernetes tests

### PR DESCRIPTION
**Pull request by agunapal**
## Description

Workflow for running nightly Kubernetes tests

Passing [run](https://github.com/pytorch/serve/actions/runs/8254810324/job/22579810555)

In addition to functionality test, this PR also checks the performance in terms of CPU usage in a k8s pod. This is important to prevent CPU throttling in a Kubernetes setup with [limits](https://www.datadoghq.com/blog/kubernetes-cpu-requests-limits/)
To do this test, we disable system gpu metrics(because there is higher CPU usage with system GPU metrics enabled) and check the cpu usage when no model is loaded in TorchServe

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Passing run is attached

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: kubernetes/tests/scripts/test_mnist.sh [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            exit 1
+        fi
+  done
+  # Get the CPU cores used by the pod
+  pod_name=$(kubectl get pods -l app=$1 -o json | jq -r '.items[].metadata.name')
+  cpu=$(kubectl top pod -l app=$1 | awk "/${pod_name}/{print \$2}")
+
+  # Check if the CPU cores exceed 2
+  if [ $(echo "$cpu" | sed 's/m$//') -gt $ACCEPTABLE_CPU_CORE_USAGE ]; then
+    echo "✘ Test failed: CPU cores $(echo "$cpu" | sed 's/m$//') for $pod_name exceeded $ACCEPTABLE_CPU_CORE_USAGE" >&2
```
Comment (msaroufim) [inline kubernetes/tests/scripts/test_mnist.sh:100]: what's the context of this PR, why is this an important test? add either a comment here or more details in the PR description

File: kubernetes/tests/scripts/test_mnist.sh [hunk 2]
```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o pipefail
+
+ACCEPTABLE_CPU_CORE_USAGE=2
+DOCKER_IMAGE=pytorch/torchserve-nightly:latest-gpu
+
+# Get relative path of example dir
```
Comment (msaroufim) [inline kubernetes/tests/scripts/test_mnist.sh:8]: crazy stuff going on here lol, worth a comment or 2

File: .github/workflows/kubernetes_tests.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: Kubernetes Nightly Tests
+
+on:
+  workflow_dispatch:
+   runs everyday  at 6:15am
+  schedule:
+    - cron:  '15 6 * * *'
+
+jobs:
+  kubernetes-tests:
```
Comment (msaroufim) [inline .github/workflows/kubernetes_tests.yml:10]: there seems to be a lot of code duplication between this workflow and https://github.com/pytorch/serve/blob/master/.github/workflows/kserve_cpu_tests.yml

File: kubernetes/tests/scripts/test_mnist.sh (hunk 1 above)
Comment (agunapal) [reply kubernetes/tests/scripts/test_mnist.sh:100]: Thanks. Added this in the description

File: kubernetes/tests/scripts/test_mnist.sh (hunk 2 above)
Comment (agunapal) [reply kubernetes/tests/scripts/test_mnist.sh:8]: lol..added

**Review by msaroufim — APPROVED** [review]

**Pull request merged**